yt/data_objects/construction_data_containers.py

```diff
@@ -899,6 +915,37 @@ def _fill_fields(self, fields):
             fi = self.ds._get_field_info(field)
             self[field] = self.ds.arr(dest, fi.units)
 
+    def _fill_sph_particles(self, fields):
+        # checks that we have the field and gets information
+        fields = [f for f in fields if f not in self.field_data]
+        if len(fields) == 0: return
+
+        ptype = self.ds._sph_ptype
+
+        for field in fields:
+            dest = np.zeros(self.ActiveDimensions, dtype="float64")
+
+            for chunk in self._data_source.chunks(fields, "io"):
+                px = chunk[(ptype,'particle_position_x')].in_units('cm')
+                py = chunk[(ptype,'particle_position_y')].in_units('cm')
+                pz = chunk[(ptype,'particle_position_z')].in_units('cm')
+                hsml = chunk[(ptype,'smoothing_length')].in_units('cm')*10.0
+                mass = chunk[(ptype,'particle_mass')].in_units('g')
+                dens = chunk[(ptype,'density')].in_units('g/cm**3')
+
+                bounds = YTArray(np.empty(6, dtype="float64"), 'cm')
+                bounds[0] = self.left_edge[0]
+                bounds[2] = self.left_edge[1]
+                bounds[4] = self.left_edge[2]
+                bounds[1] = self.right_edge[0]
+                bounds[3] = self.right_edge[1]
+                bounds[5] = self.right_edge[2]
+
+                pixelize_sph_kernel_arbitrary_grid(dest,px,py,pz,hsml,mass,
+                                                   dens,mass,bounds)
```

Comment:
hmm, did you forget to pass in the field values for the field we're smoothing onto the grid?